third_party/terraform/data_sources/data_source_iam_beta_workload_identity_pool.go.erb

```diff
@@ -0,0 +1,26 @@
+<% autogen_exception -%>
+package google
+
+<% unless version == 'ga' -%>
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceIAMBetaWorkloadIdentityPool() *schema.Resource {
+
+	dsSchema := (resourceIAMBetaWorkloadIdentityPool().Schema)
+	addRequiredFieldsToSchema(dsSchema, "workload_identity_pool_id")
+	addOptionalFieldsToSchema(dsSchema, "project")
+
+	return &schema.Resource{
+		Read:   dataSourceIAMBetaWorkloadIdentityPoolRead,
+		Schema: dsSchema,
+	}
+}
+
+func dataSourceIAMBetaWorkloadIdentityPoolRead(d *schema.ResourceData, meta interface{}) error {
+
```

Comment:
I think the test is failing because you need to do a `SetId` call right here- take a look at https://github.com/hashicorp/terraform-provider-google/blob/master/google/data_source_pubsub_topic.go for an example. The value should probably be the same as whatever SetId is for the resource.

Reply:
I've added the code to set the id, but it still fails with the same error. When I run the tests with TF_LOG set to DEBUG and can see the following steps (in that order):
- GET /v1beta/projects/magic-modules-tests/locations/global/workloadIdentityPools/bar-pool-xyw1ff8mps (seems to be the dataSource already trying to fetch the resource before it is created)
- GET fails with 404 and I can see the following logging:
  [WARN] Removing IAMBetaWorkloadIdentityPool "projects/magic-modules-tests/locations/global/workloadIdentityPools/bar-pool-xyw1ff8mps" because it's gone
  When I change the logic in dataSourceIAMBetaWorkloadIdentityPoolRead to set a different string with `d.SetId` I can see that other string appear here, which makes me believe this GET is part of the data source request
- POST /v1beta/projects/magic-modules-tests/locations/global/workloadIdentityPools?alt=json&workloadIdentityPoolId=bar-pool-xyw1ff8mps
  which is probably the logic for the normal resource creating
- polling for the create operation to complete
- GET /v1beta/projects/magic-modules-tests/locations/global/workloadIdentityPools/bar-pool-xyw1ff8mps since the resource is setup to do a GET when the operation completes as the operation does not return the resource itself
- Then immediately the errors about missing IDs:
```
...
-----------------------------------------------------
2020/10/22 16:25:20 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 200 OK
Connection: close
Transfer-Encoding: chunked
Alt-Svc: h3-Q050=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-T051=":443"; ma=2592000,h3-T050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Thu, 22 Oct 2020 14:25:20 GMT
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

e0
{
  "name": "projects/379189386656/locations/global/workloadIdentityPools/bar-pool-xyw1ff8mps",
  "displayName": "Name of pool",
  "description": "Identity pool for automated test",
  "state": "ACTIVE",
  "disabled": true
}

0


-----------------------------------------------------
2020/10/22 16:25:20 [DEBUG] Retry Transport: Stopping retries, last request was successful
2020/10/22 16:25:20 [DEBUG] Retry Transport: Returning after 1 attempts
    TestAccDataSourceIAMBetaWorkloadIdentityPool_basic: testing_new_config.go:88: no "id" found in attributes
    TestAccDataSourceIAMBetaWorkloadIdentityPool_basic: testing_new.go:57: no "id" found in attributes
--- FAIL: TestAccDataSourceIAMBetaWorkloadIdentityPool_basic (17.62s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-google-beta/google-beta	18.280s
FAIL
make: *** [testacc] Error 1
```

Reply:
Hmm, I ran the test and it passed on my end. I'm happy to merge this as-is or help investigate- whichever you prefer.